### PR TITLE
pd-ctl: decode json string in schedulers-payload output of `pd-ctl config show all`

### PR DESCRIPTION
**Pull request by wangrzneu**
### What problem does this PR solve?
Close https://github.com/pingcap/pd/issues/2367

### What is changed and how it works?

1. change the type of SchedulersPayload  from map[string]string to map[string]interface{}
2. decode config of each scheduler as an json object and save it to SchedulersPayload

### Check List

Tests

- Manual test (add detailed scripts or steps below)

1. run `pd-ctl scheduler add  balance-adjacent-region-scheduler`
2. run `pd-ctl config show all` and it will output content like below.
```
{
  "client-urls": "http://127.0.0.1:2379",
  "peer-urls": "http://127.0.0.1:2380",
  "advertise-client-urls": "http://127.0.0.1:2379",
  "advertise-peer-urls": "http://127.0.0.1:2380",
  "name": "pd-wangrzneu-VirtualBox",
  "data-dir": "/home/wangrzneu/my_git/pd/bin/default.pd-wangrzneu-VirtualBox",
  "force-new-cluster": false,
  "enable-grpc-gateway": true,
  "initial-cluster": "pd-wangrzneu-VirtualBox=http://127.0.0.1:2380",
  "initial-cluster-state": "new",
  "join": "",
  "lease": 3,
  "log": {
    "level": "",
    "format": "text",
    "disable-timestamp": false,
    "file": {
      "filename": "",
      "max-size": 0,
      "max-days": 0,
      "max-backups": 0
    },
    "development": false,
    "disable-caller": false,
    "disable-stacktrace": false,
    "disable-error-verbose": true,
    "sampling": null
  },
  "tso-save-interval": "3s",
  "metric": {
    "job": "pd-wangrzneu-VirtualBox",
    "address": "",
    "interval": "15s"
  },
  "schedule": {
    "max-snapshot-count": 3,
    "max-pending-peer-count": 16,
    "max-merge-region-size": 20,
    "max-merge-region-keys": 200000,
    "split-merge-interval": "1h0m0s",
    "enable-one-way-merge": "false",
    "enable-cross-table-merge": "false",
    "patrol-region-interval": "100ms",
    "max-store-down-time": "30m0s",
    "leader-schedule-limit": 4,
    "leader-schedule-policy": "count",
    "region-schedule-limit": 2048,
    "replica-schedule-limit": 64,
    "merge-schedule-limit": 8,
    "hot-region-schedule-limit": 4,
    "hot-region-cache-hits-threshold": 3,
    "store-balance-rate": 15,
    "tolerant-size-ratio": 0,
    "low-space-ratio": 0.8,
    "high-space-ratio": 0.7,
    "scheduler-max-waiting-operator": 5,
    "enable-remove-down-replica": "true",
    "enable-replace-offline-replica": "true",
    "enable-make-up-replica": "true",
    "enable-remove-extra-replica": "true",
    "enable-location-replacement": "true",
    "enable-debug-metrics": "false",
    "schedulers-v2": [
      {
        "type": "balance-region",
        "args": null,
        "disable": false,
        "args-payload": ""
      },
      {
        "type": "balance-leader",
        "args": null,
        "disable": false,
        "args-payload": ""
      },
      {
        "type": "hot-region",
        "args": null,
        "disable": false,
        "args-payload": ""
      },
      {
        "type": "label",
        "args": null,
        "disable": false,
        "args-payload": ""
      },
      {
        "type": "adjacent-region",
        "args": null,
        "disable": false,
        "args-payload": ""
      }
    ],
    "schedulers-payload": {
      "balance-adjacent-region-scheduler": {
        "leader-limit": 64,
        "name": "balance-adjacent-region-scheduler",
        "peer-limit": 1
      },
      "balance-hot-region-scheduler": null,
      "balance-leader-scheduler": {
        "name": "balance-leader-scheduler",
        "ranges": [
          {
            "end-key": "",
            "start-key": ""
          }
        ]
      },
      "balance-region-scheduler": {
        "name": "balance-region-scheduler",
        "ranges": [
          {
            "end-key": "",
            "start-key": ""
          }
        ]
      },
      "label-scheduler": {
        "name": "label-scheduler",
        "ranges": [
          {
            "end-key": "",
            "start-key": ""
          }
        ]
      }
    },
    "store-limit-mode": "manual"
  },
  "replication": {
    "max-replicas": 3,
    "location-labels": "",
    "strictly-match-label": "false",
    "enable-placement-rules": "false"
  },
  "pd-server": {
    "use-region-storage": "true",
    "max-gap-reset-ts": "24h0m0s",
    "key-type": "table",
    "runtime-services": "",
    "metric-storage": "",
    "dashboard-address": "http://127.0.0.1:2379"
  },
  "cluster-version": "4.1.0-alpha",
  "quota-backend-bytes": "8GiB",
  "auto-compaction-mode": "periodic",
  "auto-compaction-retention-v2": "1h",
  "TickInterval": "500ms",
  "ElectionInterval": "3s",
  "PreVote": true,
  "security": {
    "cacert-path": "",
    "cert-path": "",
    "key-path": "",
    "cert-allowed-cn": null
  },
  "label-property": {},
  "WarningMsgs": null,
  "DisableStrictReconfigCheck": false,
  "HeartbeatStreamBindInterval": "1m0s",
  "LeaderPriorityCheckInterval": "1m0s",
  "dashboard": {
    "tidb_cacert_path": "",
    "tidb_cert_path": "",
    "tidb_key_path": "",
    "public_path_prefix": "",
    "internal_proxy": false
  },
  "replication-mode": {
    "replication-mode": "majority",
    "dr-auto-sync": {
      "label-key": "",
      "primary": "",
      "dr": "",
      "primary-replicas": 0,
      "dr-replicas": 0,
      "wait-store-timeout": "1m0s",
      "wait-sync-timeout": "1m0s"
    }
  }
}

```

Side effects

- Breaking backward compatibility

### Release note

 No release note

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/pingcap/pd/v4/server/schedule"
+
```
Comment (rleungx) [inline server/server.go:32]: remove this empty line?

File: server/server.go [hunk 2]
```diff
@@ -693,9 +695,18 @@ func (s *Server) GetConfig() *config.Config {
...
-	payload := make(map[string]string)
+	payload := make(map[string]interface{})
 	for i, sche := range sches {
-		payload[sche] = configs[i]
+		var config map[string]interface{}
+		err := schedule.DecodeConfig([]byte(configs[i]), &config)
+		if err != nil {
+			log.Error("failed to decode scheduler config",
+				zap.Reflect("config", configs[i]),
+				zap.Reflect("scheduler", sche),
```
Comment (rleungx) [inline server/server.go:705]: Can we use String here?

Comment (wangrzneu) [reply server/server.go:704]: Done

File: server/server.go (hunk 1 above)
Comment (wangrzneu) [reply server/server.go]: Done

File: server/server.go (hunk 2 above)
Comment (rleungx) [reply server/server.go:704]: Actually, I mean something like
```suggestion
				zap.String("scheduler", sche),
```

Comment (wangrzneu) [reply server/server.go]: I have changed the field type, please take
 a look.

**Comment by cofyc** [comment]
In the future, I would suggest PD can treat public API as versioned API and bump the version when the schema changes or add new API instead of breaking backward compatibility. 

The old version can be deprecated first and the final removal can be made in the major version.

**Pull request merged**